android/quest/src/main/java/org/smartregister/fhircore/quest/event/EventBus.kt

```diff
@@ -27,4 +27,6 @@ class EventBus @Inject constructor() {
   val events = _events.asSharedFlow()
 
   suspend fun triggerEvent(event: AppEvent) = _events.emit(event)
+
+  fun resetReplayCache() = _events.resetReplayCache()
```

Comment:
Is it possible to have some sort of a listener to when an event has been consumed? It's easy to forget calling this function.

Reply:
This means we have to call it each time an event is consumed which is a bit repetitive

Reply:
Quick one(s), for our use case:
1. Do we expect our events to have multiple subscribers that all need to process it? 
2. Must all the events be processed or it is okay to drop some and pick the latest?
3. Do we intend to support different types of Events from this _sharedFlow_?

Reply:
To answer your questions @ndegwamartin 
1. Yes we may have such use case example listening to sync state events.
2. All event should be processed
3. Yes we already do other than `QuestionnaireSubmission` we will have other event types.

Reply:
Got it, we probably need to factor in ensuring that we initialise all subscribers before any event emissions since the replay cache only stores the last event for late subscribers